autoblocks/_impl/global_state.py

```diff
@@ -25,10 +29,19 @@ def init() -> None:
         daemon=True,
     )
     background_thread.start()
-
+    signal.signal(signal.SIGTERM, _on_unexpected_exit)
+    signal.signal(signal.SIGINT, _on_unexpected_exit)
     _started = True
 
 
+def _on_unexpected_exit(signum, frame):
```

Comment:
I wouldn't necessarily call it unexpected, sigterm is pretty standard eg in Kubernetes